react/features/app/functions.web.js

```diff
@@ -9,7 +16,53 @@ import JitsiMeetLogStorage from '../../../modules/util/JitsiMeetLogStorage';
 
 const Logger = require('jitsi-meet-logger');
 
-export * from './functions.native';
+export { _getRoomAndDomainFromUrlString } from './functions.native';
+
+/**
+ * Determines which route is to be rendered in order to depict a specific Redux
+ * store.
+ *
+ * @param {(Object|Function)} stateOrGetState - Redux state or Regux getState()
+ * method.
+ * @returns {Route}
+ */
+export function _getRouteToRender(stateOrGetState) {
+    const OS = Platform.OS;
+    const state
+        = typeof stateOrGetState === 'function'
+            ? stateOrGetState()
+            : stateOrGetState;
+
+    // If landing was shown, there is no need to show it again.
+    const { landingIsShown } = state['features/unsupported-browser'];
+    const { room } = state['features/base/conference'];
+    const component = isRoomValid(room) ? Conference : WelcomePage;
+
+    // We're using spread operator here to create copy of the route registered
+    // in registry. If we overwrite some of its properties (like 'component')
+    // they will stay unchanged in the registry.
+    const route = { ...RouteRegistry.getRouteByComponent(component) };
+
+    if ((OS === 'android' || OS === 'ios') && !landingIsShown) {
```

Comment:
The check should be whether the execution environment supports the app. The biggest piece of it is probably whether WebRTC is supported.

Only then does come the refinement that we choose to promote the mobile app in Google Chrome (if it supports WebRTC) on Android.